carmajava/signal_plugin/src/test/java/gov/dot/fhwa/saxton/carma/signal_plugin/SimpleNCVMotionPredictorTest.java

```diff
@@ -61,56 +51,67 @@ public void setUp() throws Exception {
     log.info("Setting up tests for SimpleNCVMotionPredictor");
   }
 
-  @After
-  public void tearDown() throws Exception {
-  }
-
   /**
-   * Tests the linear regression of measured vehicle position vs time.
+   * Tests the simple average speed prediction of measured vehicle position
    * @throws Exception
    */
   @Test
   public void testPredictMotion() throws Exception {
-    SimpleNCVMotionPredictor motionPredictor = new SimpleNCVMotionPredictor();
-
+	IMotionPredictor motionPredictor = new SimpleNCVMotionPredictor();
 
     // Empty list will return empty list
     List<RoutePointStamped> points = motionPredictor.predictMotion("1", new ArrayList<RoadwayObstacle>(), 0.2, 1.0);
     assertTrue(points.isEmpty());
-
-    // A single node will return an empty list
-    RoadwayObstacle n1 = newRoadwayObstacle(10.0, 0.0, 5.0);
+    
+    // A single node will return a node
+    RoadwayObstacle n1 = newRoadwayObstacle(10.0, 1.0, 1.0);
     points = motionPredictor.predictMotion("1", Arrays.asList(n1), 0.2, 1.0);
-    assertTrue(points.isEmpty());
-
+    assertEquals(6, points.size());
+    assertTrue(routePointAlmostEqual(new RoutePointStamped(10.0, 0.0, 1.0), points.get(0), 0.0001));
+    assertTrue(routePointAlmostEqual(new RoutePointStamped(10.2, 0.0, 1.2), points.get(1), 0.0001));
+    assertTrue(routePointAlmostEqual(new RoutePointStamped(10.4, 0.0, 1.4), points.get(2), 0.0001));
+    assertTrue(routePointAlmostEqual(new RoutePointStamped(10.6, 0.0, 1.6), points.get(3), 0.0001));
+    assertTrue(routePointAlmostEqual(new RoutePointStamped(10.8, 0.0, 1.8), points.get(4), 0.0001));
+    assertTrue(routePointAlmostEqual(new RoutePointStamped(11.0, 0.0, 2.0), points.get(5), 0.0001));
+    
     // 2 Node list with 0 acceleration
-    n1 = newRoadwayObstacle(10.0, 0.0, 5.0);
-    RoadwayObstacle n2 = newRoadwayObstacle(15.0, 1.0, 5.0);
-    points = motionPredictor.predictMotion("1", Arrays.asList(n1,n2), 1.0, 1.0);
-
-    assertEquals(5, points.size());
-    assertTrue(routePointAlmostEqual(new RoutePointStamped(16.0, 0.0, 1.2), points.get(0), 0.0001));
-    assertTrue(routePointAlmostEqual(new RoutePointStamped(17.0, 0.0, 1.4), points.get(1), 0.0001));
-    assertTrue(routePointAlmostEqual(new RoutePointStamped(18.0, 0.0, 1.6), points.get(2), 0.0001));
-    assertTrue(routePointAlmostEqual(new RoutePointStamped(19.0, 0.0, 1.8), points.get(3), 0.0001));
-    assertTrue(routePointAlmostEqual(new RoutePointStamped(20.0, 0.0, 2.0), points.get(4), 0.0001));
-
+    n1 = newRoadwayObstacle(14.0, 0.0, 0.8);
+    RoadwayObstacle n2 = newRoadwayObstacle(15.0, 1.0, 1.2);
```

Comment:
The comment says 2 node list with 0 acceleration, but these nodes have acceleration because it goes from 0.8 to 1.2 m/s